uportal-war/src/main/java/org/apereo/portal/layout/dlm/DistributedLayoutManager.java

```diff
@@ -908,19 +980,35 @@ private Map getPublishedChannelParametersMap(String channelPublishId) throws Por
         }
     }
 
+    @Override
     public boolean canAddNode(
             IUserLayoutNodeDescription node, String parentId, String nextSiblingId)
             throws PortalException {
         return this.canAddNode(node, this.getNode(parentId), nextSiblingId);
     }
 
     protected boolean canAddNode(
-            IUserLayoutNodeDescription node,
-            IUserLayoutNodeDescription parent,
-            String nextSiblingId)
+            final IUserLayoutNodeDescription node,
+            final IUserLayoutNodeDescription parent,
+            final String nextSiblingId)
             throws PortalException {
+
+        if ( isFragmentOwner ) return false;
```

Comment:
That's a refactor to make more explicit [a weirdness already in the code](https://github.com/Jasig/uPortal/blob/ce925e5ed28f05585fa3287b869f18bae151c405/uportal-war/src/main/java/org/apereo/portal/layout/dlm/DistributedLayoutManager.java#L952). ( [mentioned on up-dev@](https://groups.google.com/a/apereo.org/d/msg/uportal-dev/aspMtOThps8/8jrhLxlwBAAJ) ). I don't understand why `canAdd` automatically returns false for fragment owners. It might be that this is intended to be defensive because it doesn't currently check whether it's a proposal to add to the managed fragment layout or a personal layout. But it's weird, because one way or another fragment owners are actually allowed to add nodes to their layouts, as I understand it.